webapp/app/cognitoAuth.js

```diff
@@ -61,6 +63,33 @@ export const logout = () => {
   }
 }
 
+export const forgotPassword = username =>
+  new Promise((resolve, reject) => {
+    // Override Cognito error messages with a more readable one
+    if (!username) {
+      throw new Error('Please enter your email')
+    }
+
+    cognitoUser = newCognitoUser(username)
+    cognitoUser.forgotPassword(cognitoCallbacks(resolve, reject, false))
+  })
+
+export const resetPassword = (verificationCode, password) =>
+  new Promise((resolve, reject) => {
+    // Override Cognito error messages with more readable ones
+    if (!(new RegExp(/^[\S]+$/)).test(verificationCode)) {
+      throw new Error('Please enter a valid verification code')
+    } else if (!password) {
+      throw new Error('Please specify a new password')
+    } else if (!(new RegExp(/^[\S]+.*[\S]+$/)).test(password)) {
+      throw new Error('Password should not start nor end with white spaces')
```

Comment:
should we add a check to verify there's a number and an upper case letter?

Reply:
Added unique check that checks password length and content (lowercase, uppercase and numbers)